oelint_adv/cls_rule.py

```diff
@@ -96,14 +95,18 @@ def finding(self, _file, _line, override_msg=None, appendix=None):
         if _line <= 0:
             # Fix those issues, that don't come with a line
             _line = 1
+
+        _file_path = os.path.abspath(_file)
+
         if get_color():
-            if _severity == 'error':
-                return [(_line, '{file}:{color}{line}:{sev}:{id}:{msg}{reset}'.format(file=os.path.abspath(_file), color=Fore.RED, line=_line, sev=_severity, id=_display_id, msg=override_msg, reset=Style.RESET_ALL))]
-            elif _severity == 'warning':
-                return [(_line, '{file}:{color}{line}:{sev}:{id}:{msg}{reset}'.format(file=os.path.abspath(_file), color=Fore.YELLOW, line=_line, sev=_severity, id=_display_id, msg=override_msg, reset=Style.RESET_ALL))]
-            else:
-                return [(_line, '{file}:{color}{line}:{sev}:{id}:{msg}{reset}'.format(file=os.path.abspath(_file), color=Fore.GREEN, line=_line, sev=_severity, id=_display_id, msg=override_msg, reset=Style.RESET_ALL))]
-        return [(_line, '{file}:{line}:{sev}:{id}:{msg}'.format(file=os.path.abspath(_file), line=_line, sev=_severity, id=_display_id, msg=override_msg))]
+            _style = Style.RESET_ALL
+            _color = get_color_by_severity(_severity)
```

Comment:
I'm okay with these changes, but I'd propose a slightly different pattern here

```
_COLORS_BY_SEVERITY = {
    'error': Fore.RED,
    'warning': Fore.YELLOW,
    'info': Fore.GREEN
}

_color = _COLORS_BY_SEVERITY.get(_severity, '')
```

in this case we don't even have to override the default choice of the terminal for the items that shouldn't become colored.
Also this would simplify this whole operation even further